packages/esm-patient-forms-app/src/hooks/use-forms.ts

```diff
@@ -113,13 +114,15 @@ function mapToFormCompletedInfo(
 ): Array<CompletedFormInfo> {
   return allForms.map((form) => {
     const associatedEncounters = encounters.filter((encounter) => encounter.form?.uuid === form?.uuid);
-    const lastCompleted =
-      associatedEncounters.length > 0 ? new Date(associatedEncounters?.[0].encounterDatetime) : undefined;
```

Comment:
This function determines the data of the most recent matching encounter by picking the `encounterDatetime` of the first encounter and creates a new Date object from it. If there are no associated encounters, it sets `lastCompleted` to undefined. 

This implementation is not accurate. The code assumes that `associatedEncounters` is already sorted with the most recent encounter first. However, there’s no guarantee of this sorting in the code. Second, it only checks the first encounter in the array `associatedEncounters?.[0]`, which may not be the most recent encounter if the array is not sorted. 

I've modified the code to make sure we’re getting the most recent date from `associatedEncounters`. We’d then create a new Date object from the most recent encounter's date, guaranteeing that that would be the last completed date.